test/Succeed/Issue334b.agda

```diff
@@ -15,8 +17,6 @@ record E : Set where
 
 f : E -> Set
 f (mkE e unit) = f e
--- the record pattern translation does not apply to f
```

Comment:
I am bit worried that termination checking should be dependent on the record pattern translation. I think it is confusing to the user that `f` is now accepted but not if the pattern matching is turned shallow:
```agda
mutual
  f : E -> Set
  f (mkE e u) = g e u
  g : E -> Unit -> Set
  g e unit = f e
```
This is still rejected by the termination checker, as it equals
```agda
   f e = g (E.fromE e) (E.spam e)
   g e unit = f e
```
without any structural decrease.

It would be more stable to only count *non-eta* inductive constructors as increasing.

UPDATE: I think I am wrong about the record pattern translation?  It does not fire here, or?  Recursive records do not have eta, or?

Reply:
> It would be more stable to only count non-eta inductive constructors as increasing.

This is exactly what this PR does.

The record pattern translation can be safely ignored when considering termination. And yes, recursive records do not have eta.